### PR TITLE
Add sparse memory benchmark script

### PR DESCRIPTION
**Pull request by anirudh2290**
@eric-haibin-lin @reminisce 

Ran this with valgrind and massif to obtain the memory benchmarking results documented here: https://github.com/anirudh2290/mxnet/wiki/Memory-Efficiency-Benchmarking

### CODE REVIEW

File: benchmark/python/sparse/sparse_memory_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+"""Should be run with valgrind to get memory consumption
```
Comment (eric-haibin-lin) [inline benchmark/python/sparse/sparse_memory_benchmark.py]: could you add license header?

File: benchmark/python/sparse/sparse_memory_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+   for sparse format storage and dot operators"""
```
Comment (eric-haibin-lin) [inline benchmark/python/sparse/sparse_memory_benchmark.py]: nit: mention that this is for CPU mem in the doc
nit: rename file to `memory_benchmark.py`?

File: benchmark/python/sparse/sparse_memory_benchmark.py (hunk 1 above)
Comment (anirudh2290) [reply benchmark/python/sparse/sparse_memory_benchmark.py]: Added

File: benchmark/python/sparse/sparse_memory_benchmark.py (hunk 2 above)
Comment (anirudh2290) [reply benchmark/python/sparse/sparse_memory_benchmark.py]: Renamed the file to memory_benchmark.py and  added info for CPU mem

**Pull request merged**